### PR TITLE
Tidying up the WE

### PR DESCRIPTION
**Pull request by otherview**
### Why this change is needed

Part of https://github.com/obscuronet/obscuro-internal/issues/1610
This PR makes the WE closer in architecture to our other packages.

### What changes were made as part of this PR

- Using the container approach (as host and enclave) to the WE
- WE is now scoped to its business logic 
- HTTP and WS servers are moved into their own package
- Ignore the db files
- Added a few tests

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

**Review by StefanIliev545 — COMMENTED** [review]
Can we add a readme somewhere explaining the wallet extension and the code structure

File: tools/walletextension/api/server.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	// Serves the web assets for the management of viewing keys.
+	noPrefixStaticFiles, err := fs.Sub(staticFiles, staticDir)
+	if err != nil {
+		panic(fmt.Sprintf("could not serve static files. Cause: %s", err))
+	}
+	serveMux.Handle(common.PathViewingKeys, http.StripPrefix(common.PathViewingKeys, http.FileServer(http.FS(noPrefixStaticFiles))))
+
+	// Creates the actual http server
+	server := &http.Server{Addr: address, Handler: serveMux, ReadHeaderTimeout: 10 * time.Second}
```
Comment (StefanIliev545) [inline tools/walletextension/api/server.go:69]: Can we move those timeouts to constants that name what the timeout is for? And maybe leave a todo to make it configurable

**Review by StefanIliev545 — APPROVED** [review]
LGTM. Make sure to run e2e tests against.

File: go/common/httputil/httputil.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ func ExecuteHTTPReq(client *http.Client, req *http.Request) ([]byte, error) {
 	return body, nil
 }
 
-// Enables CORS to allow Obscuroscan and WalletExtension APIs to serve other web apps.
+// EnableCORS Enables CORS to allow Obscuroscan and WalletExtension APIs to serve other web apps.
```
[the remark this answers is not in the document]
Comment (otherview) [reply go/common/httputil/httputil.go:67]: Fixed 👍

File: tools/walletextension/api/server.go (hunk 1 above)
Comment (otherview) [reply tools/walletextension/api/server.go:69]: I don't see the need for this, removed 👍

File: tools/walletextension/api/routes.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+func submitViewingKeyRequestHandler(walletExt *walletextension.WalletExtension, userConn userconn.UserConn) {
+	body, err := userConn.ReadRequest()
+	if err != nil {
+		return
+	}
+
+	var reqJSONMap map[string]string
+	err = json.Unmarshal(body, &reqJSONMap)
+	if err != nil {
+		userConn.HandleError(fmt.Sprintf("could not unmarshal viewing key and signature from client to JSON: %s", err))
```
[the remark this answers is not in the document]
Comment (otherview) [reply tools/walletextension/api/routes.go:197]: Good catch, fixed 👍

**Comment by otherview** [comment]
> Can we add a readme somewhere explaining the wallet extension and the code structure

Added a comment on the readme.
The structure is the same as the ones used in host and enclave. Using containers and components to render the wallet extension.

**Comment by otherview** [comment]
> LGTM. Make sure to run e2e tests against.

Yep ! e2e ran at https://github.com/obscuronet/obscuro-test/actions/runs/5134974715

**Pull request merged**